### DISCUSSION TITLE
[Design Proposal] Dashboard Filter Smart Scoping

### DISCUSSION THREAD

**Discussion by mihir174** [apache/superset#14271] [Show and tell]
Context: 
The dashboard warning indicator was originally created to warn users when a chart might be inaccurate due to an 'incompatibility' between a filter and a chart i.e they do not reference the same dataset. Part of why this was important is because users were able to scope any filter to any chart regardless of 'compatibility'.

These designs prevent a user from scoping filters on charts that do not reference the same dataset. The goal is to prevent inaccurate charts from rendering (or not rendering) in the first place, and reducing the number of warning indicators on any given dashboard.

<img width="1945" alt="Smart Scoping (1)" src="https://user-images.githubusercontent.com/64227069/114677101-d408d500-9cbe-11eb-9b63-7ab2908dac8b.png">

**Comment by rumbin** [comment]
Will scoping a filter to a chart of a different data source still be possible? Such use cases do exist as well, e.g., for performance reasons.

**Comment by junlincc** [comment]
> Will scoping a filter to a chart of a different data source still be possible? Such use cases do exist as well, e.g., for performance reasons.

Yes!
